Compile error in bufferevent_openssl.c w/ MSVC

I am seeing a compile error building c9ec6aafb6a025f03636ae2ae7c18a2d4b8a7ed8 with openssl support.  For reasons I do not understand, my windows builds with github actions succeed (msvc and mingw).  However, msvc builds on appveyor.com are failing.  (mingw succeeds)

```
...
[ 50%] Building C object CMakeFiles/event_openssl_shared.dir/bufferevent_openssl.c.obj
"C:\Program Files\CMake\bin\cmake.exe" -E cmake_cl_compile_depends --dep-file=CMakeFiles\event_openssl_shared.dir\bufferevent_openssl.c.obj.d --working-dir=C:\projects\pvxs-dev\bundle\O.windows-x64 --filter-prefix="Note: including file: " -- C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx64\x64\cl.exe  /nologo -DHAVE_CONFIG_H -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -Devent_openssl_shared_EXPORTS -IC:\projects\pvxs-dev\bundle\O.windows-x64\include -IC:\projects\pvxs-dev\bundle\libevent\include -IC:\projects\pvxs-dev\bundle\libevent\compat -IC:\projects\pvxs-dev\bundle\libevent -IC:\projects\pvxs-dev\bundle\libevent\.\WIN32-Code -external:IC:\OpenSSL-Win64\include -external:W0 /DWIN32 /D_WINDOWS /W3 /MD /O2 /Ob2 /DNDEBUG /showIncludes /FoCMakeFiles\event_openssl_shared.dir\bufferevent_openssl.c.obj /FdCMakeFiles\event_openssl_shared.dir/ /FS -c C:\projects\pvxs-dev\bundle\libevent\bufferevent_openssl.c
bufferevent_openssl.c
C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared\ws2def.h(240): error C2011: 'sockaddr': 'struct' type redefinition
C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\winsock.h(482): note: see declaration of 'sockaddr'
...
```

The error looks like a case of including `winsock.h` before `winsock2.h`, but I can not find where this happens.  I am starting with the assumption that `winsock.h` is being included indirectly by `windows.h` for lack of `#define WIN32_LEAN_AND_MEAN`.

--- a/bufferevent_openssl.c
+++ b/bufferevent_openssl.c
@@ -24,6 +24,13 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "event2/event-config.h"
+#include "evconfig-private.h"
+
+#ifdef _WIN32
+#  include <winsock2.h>
+#endif
+
 #include <string.h>
 
 #include <openssl/ssl.h>

